[Sherman Ho] iP

nighoggDatatype opened the pull request:
Done Week 2 work. Submitting pull request

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

nighoggDatatype committed:
Done Level-1

nighoggDatatype committed:
Done Level-2

nighoggDatatype committed:
Added Task class, and cleaned up the code of warnings

nighoggDatatype committed:
Done Level-3

nighoggDatatype committed:
Refactored Task

nighoggDatatype committed:
Further refactored Task to not deal with numbering

nighoggDatatype committed:
Done A-Inheritance

nighoggDatatype committed:
Done Level-4, with some refactoring

nighoggDatatype committed:
Whoops, renamed Events to Event and actually include Event as creation. Good time for automated testing

nighoggDatatype committed:
Added test data, found bugs immediately in need of testing, unicode does not render

nighoggDatatype committed:
Somewhat fixed unicode, probably need to fix how files are compared cause line endings shouldn't matter

nighoggDatatype committed:
Removed rendundant line

nighoggDatatype committed:
Fixed another bug I missed

nighoggDatatype committed:
Added some error checking

nighoggDatatype committed:
Minor bugfix on leakage

nighoggDatatype committed:
Added Empty Argument Handling

nighoggDatatype committed:
Fixed bug to do with single arg commands

nighoggDatatype committed:
Implemented Delete

nighoggDatatype committed:
Fixed bug with event deliminator

nighoggDatatype committed:
Add minimal code to implement saving.

Code does not implement saving

Need to persist the change across runs

Let's:
* Add and call code to load and save data
* Add file safe string functions to tasks

nighoggDatatype committed:
Fix hash calculation of CheckList for better saving

nighoggDatatype committed:
Remove duke.txt from tracking

nighoggDatatype committed:
Add date time parsing to Deadline

nighoggDatatype committed:
Merge branch 'branch-Level-7'

nighoggDatatype committed:
Merge branch 'branch-Level-8'

Source branch we are merging from spreads out Date parsing too wide

This increases the possiblity for inconsistency

Let's move the parsing of strings to LocalDate into the Deadline class

nighoggDatatype committed:
Add graceful handling of badly formatted date input

nighoggDatatype committed:
Fix bug where feedback on done item was absent

nighoggDatatype committed:
Add some Class needed for the increment A-MoreOOP

nighoggDatatype committed:
Add flag to detect edits

nighoggDatatype committed:
Move some Ui code into its own class

nighoggDatatype committed:
Refactor java files into packages

Currently, the project src files are all in one single src folder.

This makes it very unclear what files are logically together and clutter the folder view.

Let's
* Create a top level package 'duke'
* Create the sub-packages 'command', 'exception' and 'task'
* Leave the files; 'Duke', 'Parser', 'TaskList', 'TaskListFileUtils' and 'Ui' in the top level package 'duke'.

Note that I only place files that will have more than one related file into packages.
I currently do not have any packages with only one file for now.
Whether that is best practice or not is unknown, and may change in the future.

nighoggDatatype committed:
Rename TaskListFileUtils to Storage for Class req

nighoggDatatype committed:
Made isDone only changable via a setter function to true.

nighoggDatatype committed:
Intergrate previously disconnected Object Oriented classes

The created object oriented classes is disconnected from the actual Duke main function

There is a requirement for OOP to be used extentively by CS2103T

Let's
* Remove ill-formatted strings due to indiscriminate package refactoring
* Properly flesh out Ui class functionality
* Intergrate prevously created class into Duke
* Edit TaskList and Storage to accomidate each other
* Edit TaskList and Ui to accomidate error printing

nighoggDatatype committed:
Allow more relaxed parsing rules

nighoggDatatype committed:
Move Error printing into Ui Class

nighoggDatatype committed:
Move last ui elements to Ui class

nighoggDatatype committed:
Add Skeleton Test Code

chewterence commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+    public static void main(String[] args) {
+        Ui ui = new Ui();
+        ui.startUpMessage();
+        TaskList store;
```
Perhaps use a more intuitive variable name here such as taskList?

chewterence commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+        }
+        Scanner in = new Scanner(System.in);
+        String line;
+        do{
```
Should the bracket and the word 'do' have a spacing between? I noticed this same issue in several other places too.

chewterence commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+            line = in.nextLine();
+            try {
+                Command c = Parser.parse(line);
+                if (c==null){ //Bye command
```
Should there be a spacing between characters in the bracket?

chewterence commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+                }
+                String data = store.run(c);
+                ui.commandMessage(c,data);
+            } catch (ParseException e) {
```
I like how you catch each exception separately, however the handling done for each of them appears to be the same function?

chewterence commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+                    }
+                }
+            }
+        }while(true);
```
Perhaps this while loop could start on a new line?

chewterence commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,71 @@
+package duke;
+
+import duke.command.*;
```
Should wildcard imports be used?

chewterence commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,71 @@
+        }
+        return c;
+    }
+    private static String[] splitTokenIntoTwo(String parseTarget,String delimiter) throws ParseException {
```
Perhaps a spacing could be used after the comma in "parseTarget, String delimiter"? I noticed similar issues in other arguments separated by commas in round brackets.

chewterence commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,71 @@
+    private static String[] splitTokenIntoTwo(String parseTarget,String delimiter) throws ParseException {
+        String[] tokens = parseTarget.split(delimiter,2);
+        if (tokens.length < 2){
+            throw new ParseException("Expected deliminter '"+ delimiter +"'", tokens[0].length());
```
Should the +'s be separated by white spaces before an inverted comma?

chewterence commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+public class Storage {
+    public static final String FILE_DIR = "data";
+    public static final String FILE_NAME = "duke.txt";
+    static int badLines = 0;//Last call bad lines
```
Should comments be indented in the same line as its code?

chewterence commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+        while(s.hasNextLine()){
+            Task t;
+            String line = s.nextLine();
+            String pattern = "([TED]),([01]),(\\d*),(.*)";
```
I like how you named the regex String as pattern 👍

chewterence commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    public static final String FILE_NAME = "duke.txt";
+    static int badLines = 0;//Last call bad lines
+
+    public static TaskList LoadTaskList() throws IOException {
```
Perhaps function names should follow camelCase?

chewterence commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+import java.util.regex.Pattern;
+
+public class Storage {
+    public static final String FILE_DIR = "data";
```
I like how you have followed screaming snake case for constant names 👍

chewterence commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+
+    public String run(Command c) throws EmptyArgumentException, BadDateArgumentException {
+        String[] args = c.run();
+        String results;
```
Should a more suitable variable name could be used here? Perhaps singular instead of plural?

chewterence commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,78 @@
+        System.out.println("Hello from\n" + logo);
+        System.out.println("No unicode allowed");
+    }
+    public void goodByeMessage(){
```
Perhaps method names should be verbs? I like how it is already in camelCase?

chewterence commented on src/main/java/duke/exception/BadDateArgumentException.java:
```diff
@@ -0,0 +1,9 @@
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
```
Perhaps constants should be in screaming snake case? I noticed the same issue at other parts of your code

chewterence commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+import java.time.format.DateTimeParseException;
+
+public class Deadline extends Task {
+    protected LocalDate by;
```
Perhaps a more intuitive variable name here?

chewterence reviewed (commented):
Generally, i find the code that you have written mostly intuitive and almost adhering relatively closely to the styling guidelines. However, i noticed and highlighted a few minor styling errors that you could perhaps fix. Do consider taking a look 👍

nighoggDatatype committed:
Add minimal Test Code to satisfy req

nicholastanvis commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,71 @@
+    }
+
+    private static String[] splitTokenIntoTwo(String parseTarget,String delimiter, String[] exception) throws ParseException{
+        List<String> exceptionList = Arrays.asList(exception);
```
Should variables that describe a collection values be named in the plural sense?

nicholastanvis commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,108 @@
+        t.setDone();
+        return formatOrderedPrint(doneIndex);
+    }
+    private String delete(int deleteIndex){
```
Should there be newlines separating these methods for better readability?

nicholastanvis commented on src/main/java/duke/command/IndexedCommand.java:
```diff
@@ -0,0 +1,13 @@
+    }
+
+    @Override
+    public String[] run() {
```
Should this method be named with a more suiting name? Perhaps a name that explains why this method has to return an array of Strings?

nicholastanvis commented on src/main/java/duke/exception/EmptyArgumentException.java:
```diff
@@ -0,0 +1,10 @@
+    /**
+     *
+     */
+    private static final long serialVersionUID = 1L;
```
Should this constant be named with all uppercase characters?

nicholastanvis commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Override
+    public String toString() {
+        return "["+ this.getStatusIcon()+"]: " +
```
I like how the string is split between multiple lines 😄

nicholastanvis commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    public static void saveTaskList(TaskList data) throws IOException {
+        List<Task> store = data.getRawData();
+        StringBuilder saveText = new StringBuilder();
+        for (Task t: store){
```
Perhaps include some whitespaces in the for statement?

nicholastanvis reviewed (commented):
Sparkling

nighoggDatatype committed:
Add Basic search framework code

nighoggDatatype committed:
Make search more intuitive

Search is currently only a wrapper on contains()

This is bad because this breaks intuition (At least my intuition) about
how searching should be done. For example: "in" should not match "Beating Minecraft"
and "minecraft" should match it, but both of those cases are unfulfilled.

Let's:
* Attempt to check description on a per word basis from the start of the word
** Allow this to be overriden by the presence of multiple spaces
* Do lowercase matching for everything by default
** Allow capital letters in the search to override this.

The reason we allow exceptions is because we sometimes want more specific searching rules.

nighoggDatatype committed:
Minor spelling correction, from overeager refactor

nighoggDatatype committed:
Add possibly incomplete Javadoc, Detailed message pending

nighoggDatatype committed:
Finish adding javadoc to every public method except main

nighoggDatatype committed:
Fix import statements

nighoggDatatype committed:
Fix spacing for code formatting

nighoggDatatype committed:
Fixed Array Formatting

nighoggDatatype committed:
Merge branch 'branch-Level-9'

nighoggDatatype committed:
Add JavaDoc for commandMessage

nighoggDatatype committed:
Merge branch 'branch-A-JavaDoc'

Added JavaDoc to new functions and made some functions private

nighoggDatatype committed:
Merge branch 'branch-A-CodingStandard'

nighoggDatatype committed:
Add missing import that somehow slipped past me

nighoggDatatype committed:
Merge remote-tracking branch 'remotes/upstream/add-gradle-support' into branch-A-Gradle

nighoggDatatype committed:
Merge branch 'branch-A-Gradle'

nighoggDatatype committed:
Add Checkstyle according to guide

nighoggDatatype committed:
Edit checkstyle to enforce SCREAMING_SNAKE_CASE for private constants

nighoggDatatype committed:
Add supressions.xml to allow gradle to work

nighoggDatatype committed:
Modify main code to satisfy automated checkStyle validation only

nighoggDatatype committed:
Merge branch 'branch-A-CheckStyle'

nighoggDatatype committed:
Address concerns in reviews for my first PR

nighoggDatatype committed:
Fix naming and bug when searching long word

nighoggDatatype committed:
Set up gradle for JavaFX

nighoggDatatype committed:
Add skeleton for JavaFx

nighoggDatatype committed:
Refactor code so that MainWindow handled output over Ui

nighoggDatatype committed:
Add Material from Tutorial to create MVP for UI

nighoggDatatype committed:
Merge branch 'branch-Level-10'

nighoggDatatype committed:
Fix checkStyle for test code

nighoggDatatype committed:
Fix minor Checkstyle issue

nighoggDatatype committed:
Fix build.gradle to make shadowJar work

nighoggDatatype committed:
Make first attempt to improve Code Quality

nighoggDatatype committed:
Use more descriptive variable names for addTask

nighoggDatatype committed:
Add Asserts for TaskList. Other classes in progress maybe

nighoggDatatype committed:
Add more asserts

nighoggDatatype committed:
Remove erronous assert

nighoggDatatype committed:
Handle out of bounds error discovered during assert creation

nighoggDatatype committed:
Add Stream and Lambda

nighoggDatatype committed:
Apply SLAP to TaskList::AddTask

nighoggDatatype committed:
Move File string generation from Storage to TaskList

nighoggDatatype committed:
Make Storage a proper object, from a static interface

nighoggDatatype committed:
Remove some completed comments and fix some Javadocs

nighoggDatatype committed:
Remove overloaded error handling function

nighoggDatatype committed:
Remove unused method

nighoggDatatype committed:
Add missing return description for javadoc

nighoggDatatype committed:
Remove unnecessary field

nighoggDatatype committed:
Add final to variables where possible

nighoggDatatype committed:
Add spacing in javadoc where appropriate

nighoggDatatype committed:
Fix typos

nighoggDatatype committed:
Improve code quality

nighoggDatatype committed:
Fix bug to do with uppercase searching

nighoggDatatype committed:
Add more final to variables

nighoggDatatype committed:
Remove redundant code discovered during testing asserts

nighoggDatatype committed:
Merge pull request #2 from nighoggDatatype/branch-A-CodeQuality

Large improvements to code quality

nighoggDatatype committed:
Merge branch 'master' into branch-A-Assertions

Add one additional assertion for bad code path

Fix at least some of the code quality issues present in branch-A-Assertions

nighoggDatatype committed:
Fix minor javadoc mangling

nighoggDatatype committed:
Merge pull request #4 from nighoggDatatype/branch-A-Assertions

Add assertions and introduce 2 bug fixes

nighoggDatatype committed:
Merge pull request #3 from nighoggDatatype/branch-A-Streams-Lambda

Add Streams and Lambdas

nighoggDatatype committed:
Add new formatting options for Deadline dates, and adjusted error messages accordingly

nighoggDatatype committed:
Fix bug where non lowercase commands with zero args were not recognized

nighoggDatatype committed:
Merge branch 'branch-BCD-Extension'

nighoggDatatype committed:
Ensure entire duke dialog can be seen at all times

nighoggDatatype committed:
Fix incorrect addition of newline when generating list in stream

nighoggDatatype committed:
Clean up messages from poor porting to GUI

nighoggDatatype committed:
Merge branch 'branch-A-BetterGui'

nighoggDatatype committed:
Add Documentation for Duke

nighoggDatatype committed:
Set theme jekyll-theme-cayman
